lfc/core/src/main/kotlin/org/lflang/generator/uc/UcMakeGenerator.kt

```diff
@@ -37,5 +41,17 @@ class UcMakeGeneratorNonFederated(private val main: Reactor, private val targetC
 
 class UcMakeGeneratorFederated(private val federate: UcFederate, targetConfig: TargetConfig, fileConfig: UcFileConfig, numEvents: Int, numReactions: Int)
     : UcMakeGenerator(federate.codeType, numEvents, numReactions) {
-    override fun generateMake(sources: List<Path>) = doGenerateMake(sources, federate.getCompileDefs())
+    override fun generateMake(sources: List<Path>): String {
+        val channelTypes = federate.interfaces.map { it.type }.toSet()
+
+        return """
+        ${"|"..doGenerateMake(sources, federate.getCompileDefs())}
+        ${channelTypes.map { when (it) {
+            NetworkChannelType.TCP_IP -> "|CFLAGS += -DNETWORK_CHANNEL_TCP_RIOT"
+            NetworkChannelType.COAP_UDP_IP -> "|CFLAGS += -DNETWORK_CHANNEL_COAP_RIOT" //TODO: Abstract RIOT away!
+            NetworkChannelType.NONE -> ""
```

Comment:
I think this will be better, once we merged the change from `NETWORK_CHANNEL_COAP_RIOT` to `NETWORK_CHANNEL_COAP`, which has been implemented in another PR I think

Reply:
I think refactoring this a little would make it clearer. See my proposal

Reply:
Very nice!